lib/drivers/medtronic600/medtronic600Simulator.js

```diff
@@ -128,6 +128,39 @@ class Medtronic600Simulator {
     }
 
     if (this.currBasal != null) {
+      // Back-fill Auto-Basal gaps
+      if (this.currBasal.scheduleName === 'Auto-Basal') {
+        const autoBasalDifference = Date.parse(event.time).valueOf() -
+          Date.parse(this.currBasal.time).valueOf();
+        // Auto-basal microboluses occur roughly every 5 minutes, but no more than 6 minutes apart.
+        // The pump doesn't tell us if the algorithm chose *not* to micro-bolus.
+        // If a 6 minute gap appears between micro-boluses, there must have been a gap.
+        if (autoBasalDifference > 360000) {
+          this.currBasal
+            .set('duration', 300000)
+            .set('basalEndTime', new Date(Date.parse(this.currBasal.time).valueOf() + 300000).toISOString());
```

Comment:
Using the Date constructor to parse dates is dangerous, as browser behaviour can change, as it recently did in Chrome 58+. Rather use `sundial.parseFromFormat`. For more details, see https://github.com/tidepool-org/chrome-uploader/pull/473